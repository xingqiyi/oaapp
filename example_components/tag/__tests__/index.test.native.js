import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Tag from '../index';

describe('Tag', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Tag>Basic</Tag>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('small size does not have closeDom', () => {
    const wrapper = shallow(<Tag small closable>Basic</Tag>);
    expect(wrapper.find('TouchableWithoutFeedback')).toHaveLength(1);
    expect(wrapper.containsMatchingElement(<Text>x</Text>)).toBeFalsy();
  });

  it('onChange then selected', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Tag onChange={onChange}>Basic</Tag>);
    expect(wrapper.state('selected')).toEqual(false);
    wrapper.find('TouchableWithoutFeedback').at(0).simulate('press');
    expect(wrapper.state('selected')).toEqual(true);
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('onClose and removed', () => {
    const onClose = jest.fn();
    const afterClose = jest.fn();
    const wrapper = shallow(<Tag closable onClose={onClose} afterClose={afterClose}>Basic</Tag>);
    expect(wrapper.find('TouchableWithoutFeedback')).toHaveLength(2);
    expect(wrapper.state('closed')).toEqual(false);
    wrapper.find('TouchableWithoutFeedback').at(1).simulate('press');
    expect(onClose).toHaveBeenCalled();
    expect(wrapper.state('closed')).toEqual(true);
    expect(afterClose).toHaveBeenCalled();
    expect(wrapper.find('TouchableWithoutFeedback')).toHaveLength(0);
  });
});
