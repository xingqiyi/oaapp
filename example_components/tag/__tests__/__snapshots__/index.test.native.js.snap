// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "borderRadius": 3,
        "flexDirection": "row",
        "overflow": "visible",
      },
      undefined,
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 3,
          "borderStyle": "solid",
          "borderWidth": 0.5,
          "overflow": "hidden",
          "paddingHorizontal": 15,
          "paddingVertical": 6,
        },
        Object {},
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#dddddd",
        },
      ]
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 12,
            "textAlign": "center",
          },
          Object {},
          Object {
            "color": "#888888",
          },
        ]
      }
    >
      Basic
       
    </Text>
  </View>
</View>
`;
